File: Add Code Here/JAVA/Sorting Leetcode Questions/MajorityElement.java
class Solution {
    
    public int majorityElement(int[] arr) {  
        
        int count = 1;
        int ans =0;

        for (int i = 0; i < arr.length; i++) {
            if (arr[i]==arr[ans]){
                count++;
            }else {
                count--;
            }

            if (count==0){
                ans = i;
                count=1;
            }
        }
        return arr[ans];
   }
}

File: Add Code Here/C++/tree/Majority_element.cpp
#include <bits/stdc++.h>
using namespace std;
void findMajority(int arr[], int n)
{
	int maxCount = 0;
	int index = -1; // sentinels
	for (int i = 0; i < n; i++) {
		int count = 0;
		for (int j = 0; j < n; j++) {
			if (arr[i] == arr[j])
				count++;
		}
		if (count > maxCount) {
			maxCount = count;
			index = i;
		}
	}
	if (maxCount > n / 2)
		cout << arr[index] << endl;

	else
		cout << "No Majority Element" << endl;
}
int main()
{
	int arr[] = { 1, 1, 2, 1, 3, 5, 1 };
	int n = sizeof(arr) / sizeof(arr[0]);

	// Function calling
	findMajority(arr, n);

	return 0;
}
